src/application/can_handler/can_handler.c

```diff
@@ -0,0 +1,80 @@
+#include "can_handler.h"
+
+QueueHandle_t busQueue_rx;
+QueueHandle_t busQueue_tx;
+
+can_callback_t callback_map[30];
+
+void can_handle_rx_isr(const can_msg_t *message, uint32_t timestamp)
+{
+    // The timestamp parameter passed to the handler is some internal FDCAN thing that I don't know how to convert to a sensible value
+    // Just use millis_() for now
+    xQueueSend(busQueue_rx, &message, 10);
+    return;
+}
+
+w_status_t can_register_callback(can_msg_type_t msg_type, can_callback_t callback)
+{
+    callback_map[msg_type] = callback;
+    return W_SUCCESS;
+}
+
+void can_handler_tx(const can_msg_t *message)
+{
+    if (xQueueSend(busQueue_tx, message, 10))
+    {
+        return W_SUCCESS;
+    }
+    else
+    {
+        return W_FAILURE;
+    }
+}
+
+void can_handler_task_rx(void *argument)
+{
+    for (;;)
+    {
+        can_msg_t *rx_msg;
+        if (xQueueReceive(busQueue_rx, &rx_msg, 10) == pdTRUE)
+        {
+            uint16_t msgType = get_message_type(rx_msg);
+            if (callback_map[msgType] != NULL)
+            {
+                callback_map[msgType](rx_msg);
+            }
+        }
+    }
+}
+
+void can_handler_task_tx(void *argument)
+{
+    for (;;)
+    {
+        can_msg_t tx_msg;
+        // Block the thread until we see data in the bus queue or 1 sec elapses
+        if (xQueueReceive(busQueue_tx, &tx_msg, 1000) == pdTRUE)
+        { // Returns pdTRUE if we got a message, pdFALSE if timed out
+            if (can_send(&tx_msg) == false)
+            {
+                logError("CAN", "CAN send failed!");
```

Comment:
```suggestion
                logText("CAN", "CAN send failed!");
```
the new function is called `logText`